packages/nimble-components/src/theme-provider/tests/theme-provider.spec.ts

```diff
@@ -16,6 +19,122 @@ describe('Theme Provider', () => {
         );
     });
 
+    describe('lang token', () => {
+        async function setup(
+            langValue: string | undefined
+        ): Promise<Fixture<ThemeProvider>> {
+            return fixture<ThemeProvider>(
+                html`<${themeProviderTag} ${
+                    langValue === undefined ? '' : `lang="${langValue}"`
+                }>
+                    </${themeProviderTag}>`
+            );
+        }
+
+        let element: ThemeProvider;
+        let connect: () => Promise<void>;
+        let disconnect: () => Promise<void>;
+        let pageLangToRestore: string | undefined;
+
+        beforeEach(() => {
+            pageLangToRestore = document.documentElement.lang;
```

Comment:
When accessing the property value you can't distinguish between an absent lang and a lang of empty string. The current code is using a truthiness check to assume that a lang of empty string means remove the lang but it can't actually accurately reflect the state of what's configured.

I'd avoid all the mental gymnastics and switching between manipulating the property and the attribute and just use the attribute consistently in this case.

```
pageLangToRestore = document.documentElement.getAttribute('lang');
if (pageLangToRestore === null) {
    document.documentElement.removeAttribute('lang');
} else {
    document.documentElement.setAttribute('lang', pageLangToRestore);
}

```